### ISSUE TITLE
Unexpected nullable warning when using field keyword with null coalescing operator

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#84719]
```csharp
class C
{
    private string? _s;

    public string S
    {
        get => _s ?? string.Empty;
        set => _s = value;
    }

    
    public string S2
    {
        get => field ?? string.Empty;
        set => field = value;
    }
}
```

This produces the following:

```
warning CS9264: Non-nullable property 'S2' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier, or declaring the property as nullable, or adding '[field: MaybeNull, AllowNull]' attributes.
```

However, both S and S2 are actually equivalent.

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/76662

**Comment by Youssef1313** [comment]
Looks fixed. I was testing something on SharpLab which was very outdated. Not reproducible in https://lab.razor.fyi/
